### PR TITLE
planner: fix can not found column bug

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

In `(p *PhysicalHashJoin) attach2TaskForMpp(tasks ...task)`, the task maybe changed, so we should use the updated task when construct the final result task.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Fix `can not found column in Schema column` error for mpp queries
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -747,6 +747,54 @@ func (s *testIntegrationSerialSuite) TestMPPShuffledJoin(c *C) {
 		res.Check(testkit.Rows(output[i].Plan...))
 	}
 }
```
Comment (fzhedu) [inline planner/core/integration_test.go:749]: add en empty line.

File: planner/core/task.go [hunk 2]
```diff
@@ -782,7 +782,12 @@ func (p *PhysicalHashJoin) attach2TaskForMpp(tasks ...task) task {
 	lCost := lTask.cost()
 	rCost := rTask.cost()
 
-	outerTask := tasks[1-p.InnerChildIdx].(*mppTask)
+	// can not use the task from tasks because it maybe updated.
+	outerTaskIndex := 1 - p.InnerChildIdx
```
Comment (fzhedu) [inline planner/core/task.go:786]: does this happen in `convertPartitionKeysIfNeed`?

Comment (windtalker) [reply planner/core/task.go:786]: No, `convertPartitionKeysIfNeed` returns updated mpp task directly.

**Review by hanfei1991 — APPROVED** [review]

File: planner/core/exhaust_physical_plans.go [hunk 3]
```diff
@@ -1886,8 +1886,22 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		lPartitionKeys, rPartitionKeys := p.GetPotentialPartitionKeys()
 		if prop.MPPPartitionTp == property.HashType {
 			var matches []int
-			if matches = prop.IsSubsetOf(lPartitionKeys); len(matches) == 0 {
+			if p.JoinType == InnerJoin {
```
Comment (fzhedu) [inline planner/core/exhaust_physical_plans.go:1889]: do previous tests not cover these cases?

Comment (windtalker) [reply planner/core/exhaust_physical_plans.go:1889]: I guess so.

**Review by fzhedu — APPROVED** [review]
/LGTM

**Pull request merged**